File: crates/hydra-tui/internal/theme/palette.go
package theme

import "github.com/charmbracelet/lipgloss"

// Hydra brand palette — 7 semantic colors + utility
var (
	HydraBlue       = lipgloss.Color("#6495ED") // Primary accent, borders, headers
	HydraCyan       = lipgloss.Color("#00D2D2") // Username, keywords, sister working
	HydraGreen      = lipgloss.Color("#50C878") // Success, connected, healthy
	HydraRed        = lipgloss.Color("#DC5050") // Error, offline, critical
	HydraYellow     = lipgloss.Color("#F0C850") // Warning, partial, uncertain
	HydraOrange     = lipgloss.Color("#F0A03C") // Decide phase, action needed
	HydraPurple     = lipgloss.Color("#A078DC") // Model name, Learn phase
	HydraDim        = lipgloss.Color("#808080") // Labels, paths, subtle text
	HydraBorder     = lipgloss.Color("#808080") // Inactive borders
	HydraBorderActive = lipgloss.Color("#6495ED") // Active = HYDRA_BLUE

	// Diff backgrounds
	DiffRedBG  = lipgloss.Color("#501414")
	DiffGreenBG = lipgloss.Color("#143C14")
)

// DotCategory represents the type of activity for colored dots.
type DotCategory int

const (
	DotThinking DotCategory = iota
	DotSisterWorking
	DotSuccess
	DotError
	DotWarning
	DotCognitive
	DotApproval
)

// DotColor returns the color for a given dot category.
func DotColor(cat DotCategory) lipgloss.Color {
	switch cat {
	case DotThinking:
		return HydraDim
	case DotSisterWorking:
		return HydraCyan
	case DotSuccess:
		return HydraGreen
	case DotError:
		return HydraRed
	case DotWarning:
		return HydraYellow
	case DotCognitive:
		return HydraPurple
	case DotApproval:
		return HydraOrange
	default:
		return HydraDim
	}
}

// ConfidenceColor returns color for belief confidence levels.
func ConfidenceColor(confidence float64) lipgloss.Color {
	if confidence > 0.85 {
		return HydraGreen
	} else if confidence >= 0.5 {
		return HydraYellow
	}
	return HydraRed
}

// HealthColor returns color for health percentage.
func HealthColor(pct float64) lipgloss.Color {
	if pct >= 90 {
		return HydraGreen
	} else if pct >= 50 {
		return HydraYellow
	}
	return HydraRed
}

// PhaseColor returns color for cognitive phases.
func PhaseColor(phase string) lipgloss.Color {
	switch phase {
	case "Perceive":
		return HydraBlue
	case "Think":
		return HydraYellow
	case "Decide":
		return HydraOrange
	case "Act":
		return HydraGreen
	case "Learn":
		return HydraPurple
	default:
		return HydraDim
	}
}

// Spinner phases for Hydra-themed spinner
var SpinnerChars = []rune{'◌', '◐', '◑', '◒', '◓', '●'}

File: crates/hydra-tui/src/theme/palette.rs
use ratatui::style::Color;

// Hydra brand palette — 7 semantic colors + utility
pub const HYDRA_BLUE: Color = Color::Rgb(100, 149, 237); // Primary accent, borders, headers
pub const HYDRA_CYAN: Color = Color::Rgb(0, 210, 210); // Username, keywords, sister working
pub const HYDRA_GREEN: Color = Color::Rgb(80, 200, 120); // Success, connected, healthy
pub const HYDRA_RED: Color = Color::Rgb(220, 80, 80); // Error, offline, critical
pub const HYDRA_YELLOW: Color = Color::Rgb(240, 200, 80); // Warning, partial, uncertain
pub const HYDRA_ORANGE: Color = Color::Rgb(240, 160, 60); // Decide phase, action needed
pub const HYDRA_PURPLE: Color = Color::Rgb(160, 120, 220); // Model name, Learn phase
pub const HYDRA_DIM: Color = Color::DarkGray; // Labels, paths, subtle text
pub const HYDRA_BG: Color = Color::Reset; // Terminal default
pub const HYDRA_FG: Color = Color::Reset; // Terminal default
pub const HYDRA_BORDER: Color = Color::DarkGray; // Inactive borders
pub const HYDRA_BORDER_ACTIVE: Color = Color::Rgb(100, 149, 237); // Active = HYDRA_BLUE

// Diff backgrounds
pub const DIFF_RED_BG: Color = Color::Rgb(80, 20, 20);
pub const DIFF_GREEN_BG: Color = Color::Rgb(20, 60, 20);

/// Dot color by activity category
pub fn dot_color(category: DotCategory) -> Color {
    match category {
        DotCategory::Thinking => HYDRA_DIM,
        DotCategory::SisterWorking => HYDRA_CYAN,
        DotCategory::Success => HYDRA_GREEN,
        DotCategory::Error => HYDRA_RED,
        DotCategory::Warning => HYDRA_YELLOW,
        DotCategory::Cognitive => HYDRA_PURPLE,
        DotCategory::Approval => HYDRA_ORANGE,
    }
}

#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum DotCategory {
    Thinking,
    SisterWorking,
    Success,
    Error,
    Warning,
    Cognitive,
    Approval,
}

/// Confidence bracket color for belief borders
pub fn confidence_color(confidence: f64) -> Color {
    if confidence > 0.85 {
        HYDRA_GREEN
    } else if confidence >= 0.5 {
        HYDRA_YELLOW
    } else {
        HYDRA_RED
    }
}

/// Health percentage color
pub fn health_color(pct: f64) -> Color {
    if pct >= 90.0 {
        HYDRA_GREEN
    } else if pct >= 50.0 {
        HYDRA_YELLOW
    } else {
        HYDRA_RED
    }
}

/// Cognitive phase color
pub fn phase_color(phase: &str) -> Color {
    match phase {
        "Perceive" => HYDRA_BLUE,
        "Think" => HYDRA_YELLOW,
        "Decide" => HYDRA_ORANGE,
        "Act" => HYDRA_GREEN,
        "Learn" => HYDRA_PURPLE,
        _ => HYDRA_DIM,
    }
}
